errors/errors.go

```diff
@@ -0,0 +1,15 @@
+package errors
+
+import "fmt"
+
+// Error allows us to create constant errors
```

Comment:
I'm not sure if this error file implements the  [same](https://dave.cheney.net/2016/04/07/constant-errors) concepts as you referred to earlier; but I think if it is, we should put the link to the blog post in the code in a comment for those who are new to Go / interested. 

But, I also might just be projecting my own lack of understanding of errors so far into the code. What do you think?